api/v3/utils.php

```diff
@@ -461,6 +461,382 @@ function _civicrm_api3_store_values(&$fields, &$params, &$values) {
 /**
  * Get function for query object api.
  *
+ * This is a simple get function, but it should be usable for any kind of
+ * entity. I created it to work around CRM-16036.
+ *
+ * @param string $dao_name
+ *   Name of DAO
+ * @param array $params
+ *   As passed into api get function.
+ * @param bool $isFillUniqueFields
+ *   Do we need to ensure unique fields continue to be populated for this api? (backward compatibility).
+ * @param array $extraMysql
```

Comment:
Fantastic - I think there is actually some code that is now by-passed in there too - but it was a bit of a crazy mission to get through the tests. Like I say though - I don't think you can get any errors past them so any changes now that don't cause test fails should be really safe